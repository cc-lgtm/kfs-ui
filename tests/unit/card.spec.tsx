import { mount } from '@vue/test-utils'
import CcCard from '@/packages/card/index'

test('test card html', () => {
  const wraper = mount(CcCard)

  expect(wraper.html).toMatchSnapshot()
})

const testCard = {
  template: `<CcCard value="value">{{ count }}<CcCard/>`,
  props: ['value'],
  data() {
    return {
      count: '0'
    }
  }
}

test('test card text', () => {
  const wrapwer = mount(testCard, {
    props: {
      value: 'test'
    }
  })

  expect(wrapwer.text()).toContain('0')
})

test('test card slots && click', () => {
  const wrapwer = mount(testCard, {
    props: {
      value: ''
    }
  })

  wrapwer.trigger('click')
  expect(wrapwer.emitted()).toHaveProperty('click')
  expect(wrapwer.text()).toContain(0)
})


