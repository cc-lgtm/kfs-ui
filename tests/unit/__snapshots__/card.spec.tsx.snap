// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test card html 1`] = `[Function]`;
